code/modules/mob/living/silicon/mommi/emote.dm

```diff
@@ -1,247 +1,11 @@
-			message = "<B>[src]</B> shudders violently for a moment, then becomes motionless, its eyes slowly darkening."
-			m_type = VISIBLE
-
-		if ("glare")
-			var/M = null
-			if (param)
-				for (var/mob/A in view(null, null))
-					if (param == A.name)
-						M = A
-						break
-			if (!M)
-				param = null
-
-			if (param)
-				message = "<B>[src]</B> glares at [param] as best a robot spider can glare."
-			else
-				message = "<B>[src]</B> glares as best a robot spider can glare."
-
-		if ("stare")
-			var/M = null
-			if (param)
-				for (var/mob/A in view(null, null))
-					if (param == A.name)
-						M = A
-						break
-			if (!M)
-				param = null
-
-			if (param)
-				message = "<B>[src]</B> stares at [param]."
-			else
-				message = "<B>[src]</B> stares."
-
-		if ("look")
-			var/M = null
-			if (param)
-				for (var/mob/A in view(null, null))
-					if (param == A.name)
-						M = A
-						break
-
-			if (!M)
-				param = null
-
-			if (param)
-				message = "<B>[src]</B> looks at [param]."
-			else
-				message = "<B>[src]</B> looks."
-			m_type = VISIBLE
-
-		if("beep")
-			var/M = null
-			if(param)
-				for (var/mob/A in view(null, null))
-					if (param == A.name)
-						M = A
-						break
-			if(!M)
-				param = null
-
-			if (param)
-				message = "<B>[src]</B> beeps at [param]."
-			else
-				message = "<B>[src]</B> beeps."
-			playsound(src, 'sound/machines/twobeep.ogg', 50, 0)
-			m_type = VISIBLE
-
-		if("ping")
-			var/M = null
-			if(param)
-				for (var/mob/A in view(null, null))
-					if (param == A.name)
-						M = A
-						break
-			if(!M)
-				param = null
-
-			if (param)
-				message = "<B>[src]</B> pings at [param]."
-			else
-				message = "<B>[src]</B> pings."
-			playsound(src, 'sound/machines/ping.ogg', 50, 0)
-			m_type = VISIBLE
-
-		if("buzz")
-			var/M = null
-			if(param)
-				for (var/mob/A in view(null, null))
-					if (param == A.name)
-						M = A
-						break
-			if(!M)
-				param = null
-
-			if (param)
-				message = "<B>[src]</B> buzzes at [param]."
-			else
-				message = "<B>[src]</B> buzzes."
-			playsound(src, 'sound/machines/buzz-sigh.ogg', 50, 0)
-			m_type = VISIBLE
-
-		if("comment")
-			var/M = null
-			if(param)
-				for (var/mob/A in view(null, null))
-					if (param == A.name)
-						M = A
-						break
-			if(!M)
-				param = null
-
-			if (param)
-				message = "<B>[src]</B> cheerily vocalizes at [param]."
-			else
-				message = "<B>[src]</B> vocalizes."
-			playsound(src, get_sfx("mommicomment"),50, 0)
-			m_type = VISIBLE
-
-		else
-//			to_chat(custom_emote(VISIBLE, act) src, text("Invalid Emote: [], use *help", act))
-	if ((message && src.stat == 0))
-		if (m_type & 1)
-			for(var/mob/O in viewers(src, null))
-				O.show_message(message, m_type)
-		else
-			for(var/mob/O in hearers(src, null))
-				O.show_message(message, m_type)
-	return
\ No newline at end of file
+/datum/emote/sound/silicon/mommi/comment/run_emote(mob/user, params, type_override)
+	sound = get_sfx("mommicomment")
```

Comment:
don't you have to call the parent?
Or even just assign this in New and pick from the list in a more generic `run_emote`

Reply:
I think both options are equivalent and this is the cleanest one.
Although I did need to call the parent here